### ISSUE TITLE
ha_proxy information is not very clear

### ISSUE THREAD

**Issue by jadekler** [cloudfoundry/docs-deploying-cf#143]
Relevant doc: https://docs.cloudfoundry.org/deploying/openstack/cf-stub.html

> 	Replace RSA_PRIVATE_KEY and SSL_CERTIFICATE_SIGNED_BY_PRIVATE_KEY with the PEM
> encoded private key and certificate associated with the system domain and apps domains you’ve
> configured to terminate at the floating IP associated with the ha_proxy job.

This is not very clear. Replace xxx with the xxx associated with the xxx ... at the floating IP associated with xxx. Is there a way to be more clear about this? Perhaps just 'Replace xxx with xxx'?

**Comment by pspinrad** [comment]
Yeah, that's bad-- will clarify-- thanks for pointing it out!

**Comment by pspinrad** [comment]
@voelzmo you configure a key and cert for HAproxy in the CF manifest on Openstack -- this is the main key and cert pair used to access CF, right?

For people using the relevant doc linked above, where can we tell them to find these values, or in what previous step did they first generate or choose them?  Thanks!

**Comment by voelzmo** [comment]
@pspinrad @jadekler The situation is unfortunately a bit complicated:
* You usually setup your CF installation with some kind of domain you own, e.g. `pivotal.io`
* You define separate subdomains for your applications and system services, e.g. `apps.pivotal.io` and `sys.pivotal.io`
* You have a loadbalancer in front (in the OpenStack case this is HAProxy, in the AWS case this is an AWS ELB) which does SSL termination. This loadbalancer gets assigned a floating IP so it can be accessed from the outside
* With means outside of CF and BOSH you need to issue a certificate for a wildcard domain `*.pivotal.io` and have it signed by some Certificate Authority (i.e. a private key)
* The private key and the certificate signed with that key need to be put into those variables mentioned

So TL;DR: The Certificate and private key necessary here are not created during the installation of CF, but are a prerequisite. Probably we should add a section about prerequisites somewhere and reference this section here.

PS:  Instructions for vSPhere have essentially the same problem but slightly different wording: https://docs.cloudfoundry.org/deploying/common/vsphere-vcloud-cf-stub.html#editing

**Comment by pspinrad** [comment]
@voelzmo - great info, thanks for laying it out!

**Comment by jadekler** [comment]
Thanks @voelzmo
